src/engine/Server.cpp

```diff
@@ -14,25 +14,27 @@
 #include "engine/ExportQueryExecutionTrees.h"
 #include "engine/QueryPlanner.h"
 #include "util/BoostHelpers/AsyncWaitForFuture.h"
+#include "util/MemorySize/MemorySize.h"
 #include "util/OnDestructionDontThrowDuringStackUnwinding.h"
 #include "util/http/websocket/MessageSender.h"
 
 template <typename T>
 using Awaitable = Server::Awaitable<T>;
 
 // __________________________________________________________________________
-Server::Server(unsigned short port, int numThreads, size_t maxMemGB,
-               std::string accessToken, bool usePatternTrick)
+Server::Server(unsigned short port, int numThreads,
+               ad_utility::MemorySize maxMemGB, std::string accessToken,
+               bool usePatternTrick)
     : numThreads_(numThreads),
       port_(port),
       accessToken_(std::move(accessToken)),
-      allocator_{
-          ad_utility::makeAllocationMemoryLeftThreadsafeObject(maxMemGB *
-                                                               (1ULL << 30U)),
-          [this](size_t numBytesToAllocate) {
-            cache_.makeRoomAsMuchAsPossible(MAKE_ROOM_SLACK_FACTOR *
-                                            numBytesToAllocate / sizeof(Id));
-          }},
+      allocator_{ad_utility::makeAllocationMemoryLeftThreadsafeObject(
+                     maxMemGB.getBytes()),
```

Comment:
This object can also directly use the `MemorySize`, but that should then be a separate PR.

Reply:
Already done, but that one is based on https://github.com/ad-freiburg/qlever/pull/1067 , which makes it a bit useless.